web/forms.py

```
@@ -853,38 +854,96 @@
     captcha = CaptchaField(
         widget=TailwindCaptchaTextInput(
             attrs={"class": "block w-full border rounded p-2 focus:outline-none focus:ring-2 focus:ring-indigo-500"}
         )
     )
 
+    def clean_course_title(self):
+        """Validate and clean the course_title field."""
+        title = self.cleaned_data.get("course_title")
+        # Validate title contains only valid characters
+        if not re.match(r"^[\w\s-]+$", title):
+            raise forms.ValidationError("Title can only contain letters, numbers, spaces, and hyphens")
```

Ruff (0.8.2)

Avoid specifying long messages outside the exception class

(TRY003)